src/gpu/data/matrix.cu

```diff
@@ -1,9 +1,36 @@
 #include "matrix.cuh"
 #include <algorithm>
 #include <thrust/inner_product.h>
+#include <thrust/extrema.h>
 
 namespace tsvd
 {
+	void max_index_per_column(Matrix<tsvd_float>& A, std::vector<int>& result_array, DeviceContext& context){
+
+		int result;
+		for (int i=0; i<A.columns(); i++) {
+			safe_cublas(cublasIsamax(context.cublas_handle, A.rows(), A.data() + i*A.rows(), 1, &result));
+			if (i == 0) {
+				result_array[i] = result - 1;
+			} else  {
+				result_array[i] = result - 1 + A.rows();
```

Comment:
Shouldn't this be `result_array[i] = result - 1 + i * A.rows();`?

`cublasIsamax` will always return a value in range `[ 0, A.rows() )` since we're passing a start pointer `A.data() + i * A.rows()` and the size `A.rows()`? Wouldn't `result - 1 + A.rows();` in such a case always return an index corresponding to the 2nd column (value in `[ 0, A.rows() )` plus `A.rows()`)?

If so, we can change that whole if/else to simply `result_array[i] = result - 1 + i * A.rows();`.

Also: is `contex.cublas_handle`, thread safe? Could we for example run:
```
#pragma omp parallel for
for (int i=0; i<A.columns(); i++) {
// loop content
}
```

Or are the loop content calculated so fast we wouldn't see a difference (or even a slowdown)?